server/statistics/topn.go

```diff
@@ -24,27 +24,31 @@ import (
 type TopNItem interface {
 	// ID is used to check identity.
 	ID() uint64
-	// Less tests whether the current item is less than the given argument.
-	Less(then TopNItem) bool
+	// Less tests whether the current item is less than the given argument in the `k`th dimension.
+	Less(k int, than TopNItem) bool
 }
 
-// TopN maintains the N largest items.
+// TopN maintains the N largest items of multiple dimensions.
 type TopN struct {
 	rw     sync.RWMutex
-	n      int
-	topn   *indexedHeap
-	rest   *indexedHeap
+	topns  []*singleTopN
 	ttlLst *ttlList
 }
 
-// NewTopN returns a TopN with given TTL.
-func NewTopN(n int, ttl time.Duration) *TopN {
-	return &TopN{
-		n:      maxInt(n, 1),
-		topn:   newTopNHeap(n),
-		rest:   newRevTopNHeap(n),
+// NewTopN returns a k-dimensional TopN with given TTL.
+// NOTE: panic if k <= 0 or n <= 0.
+func NewTopN(k, n int, ttl time.Duration) *TopN {
+	if k <= 0 || n <= 0 {
+		panic("invalid arguments for NewTopN")
```

Comment:
How about printing the value of `n` and `k`?

Reply:
Fixed.